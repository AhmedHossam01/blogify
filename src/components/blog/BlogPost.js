import React, { Component } from "react";
import Card from "react-bootstrap/Card";
import Button from "react-bootstrap/Button";
import AuthorCard from "./AuthorCard";

export default class BlogPost extends Component {
  render() {
    const { author, title, body, picture } = this.props;
    return (
      <Card style={{ width: "18rem" }}>
        <Card.Img variant="top" src={picture} />
        <Card.Body>
          <Card.Title>{title}</Card.Title>
          <Card.Text>{body}</Card.Text>
          <AuthorCard author={author}></AuthorCard>
          <Button variant="dark" bg="dark">
            Read More
          </Button>
        </Card.Body>
      </Card>
    );
  }
}
