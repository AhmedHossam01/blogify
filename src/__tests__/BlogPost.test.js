import React from "react";
import { render } from "react-dom";
import { act } from "react-dom/test-utils";
import BlogPost from "../components/blog/BlogPost";

describe("BlogPost component", () => {
  it("should render with assigned data", () => {
    const fakePost = {
      title: "Lorem Ipsum",
      id: 2,
      body: "Lorem Ipsum Ist dolor sit"
    };

    act(() => {
      const container = document.createElement("div");
      render(
        <BlogPost
          key={fakePost.id}
          title={fakePost.title}
          body={fakePost.body}
        ></BlogPost>,
        container
      );
      expect(container).toMatchSnapshot();
    });
  });
});
