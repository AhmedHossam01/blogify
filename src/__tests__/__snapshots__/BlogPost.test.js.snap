// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPost component should render with assigned data 1`] = `
<div>
  <div
    class="card"
    style="width: 18rem;"
  >
    <img
      class="card-img-top"
    />
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
      >
        Lorem Ipsum
      </div>
      <p
        class="card-text"
      >
        Lorem Ipsum Ist dolor sit
      </p>
      <button
        bg="dark"
        class="btn btn-dark"
        type="button"
      >
        Read More
      </button>
    </div>
  </div>
</div>
`;
